crates/node/core/src/args/pruning.rs

```diff
@@ -3,48 +3,248 @@
+    pub transaction_lookup_before: Option<BlockNumber>,
+
+    // Receipts
+    /// Prunes all receipt data.
+    #[arg(long = "prune.receipts.full", conflicts_with_all = &["prune.receipts.distance", "prune.receipts.before"])]
+    pub receipts_full: bool,
+    /// Prune receipts before the `head-N` block number. In other words, keep last N + 1 blocks.
+    #[arg(long = "prune.receipts.distance", value_name = "BLOCKS", conflicts_with_all = &["prune.receipts.full", "prune.receipts.before"])]
+    pub receipts_distance: Option<u64>,
+    /// Prune receipts before the specified block number. The specified block number is not pruned.
+    #[arg(long = "prune.receipts.before", value_name = "BLOCK_NUMBER", conflicts_with_all = &["prune.receipts.full", "prune.receipts.distance"])]
+    pub receipts_before: Option<BlockNumber>,
+
+    // Account History
+    /// Prunes all account history.
+    #[arg(long = "prune.accounthistory.full", conflicts_with_all = &["prune.accounthistory.distance", "prune.accounthistory.before"])]
+    pub account_history_full: bool,
+    /// Prune account before the `head-N` block number. In other words, keep last N + 1 blocks.
+    #[arg(long = "prune.accounthistory.distance", value_name = "BLOCKS", conflicts_with_all = &["prune.accounthistory.full", "prune.accounthistory.before"])]
+    pub account_history_distance: Option<u64>,
+    /// Prune account history before the specified block number. The specified block number is not
+    /// pruned.
+    #[arg(long = "prune.accounthistory.before", value_name = "BLOCK_NUMBER", conflicts_with_all = &["prune.accounthistory.full", "prune.accounthistory.distance"])]
+    pub account_history_before: Option<BlockNumber>,
+
+    // Storage History
+    /// Prunes all storage history data.
+    #[arg(long = "prune.storagehistory.full", conflicts_with_all = &["prune.storagehistory.distance", "prune.storagehistory.before"])]
+    pub storage_history_full: bool,
+    /// Prune storage history before the `head-N` block number. In other words, keep last N + 1
+    /// blocks.
+    #[arg(long = "prune.storagehistory.distance", value_name = "BLOCKS", conflicts_with_all = &["prune.storagehistory.full", "prune.storagehistory.before"])]
+    pub storage_history_distance: Option<u64>,
+    /// Prune storage history before the specified block number. The specified block number is not
+    /// pruned.
+    #[arg(long = "prune.storagehistory.before", value_name = "BLOCK_NUMBER", conflicts_with_all = &["prune.storagehistory.full", "prune.storagehistory.distance"])]
+    pub storage_history_before: Option<BlockNumber>,
+
+    // Receipts Log Filter
+    /// Configure receipts log filter. Format:
+    /// <`address`>:<`prune_mode`>[,<`address`>:<`prune_mode`>...] Where <`prune_mode`> can be
+    /// 'full', 'distance:<`blocks`>', or 'before:<`block_number`>'
+    #[arg(long = "prune.receiptslogfilter", value_name = "FILTER_CONFIG", value_delimiter = ',')]
```

Comment:
```suggestion
    #[arg(long = "prune.receiptslogfilter", value_name = "FILTER_CONFIG", value_delimiter = ',', value_parser = PruningArgs::parse_receipts_log_filter)]
```

this way you shouldn't have to call `PruningArgs::parse_receipts_log_filter` explicitly in `PruningArgs::prune_config`

Reply:
It doesn't look like simply doing `value_parser = PruningArgs::parse_receipts_log_filter` works out of the box. IIUC I would need to create a new Parser struct and do something like 

```
impl TypedValueParser for ReceiptsLogFilterParser
```

where the impl seems quite involved. I'd prefer to just keep the explicit call to `self.parse_receipts_log_filter` instead as a more imperative style.

Reply:
check out how it's done here

https://github.com/paradigmxyz/reth/blob/d5906f3f89653059e37afff71a9913a9b43a5039/crates/node/core/src/args/payload_builder.rs#L75-L100

or here

https://github.com/paradigmxyz/reth/blob/d5906f3f89653059e37afff71a9913a9b43a5039/crates/cli/commands/src/db/get.rs#L193-L200

Reply:
Addressed in https://github.com/paradigmxyz/reth/pull/10639/commits/47eeb16ed5274df6263bdafde15ae5c1e2b85fe4